java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java

```diff
@@ -133,7 +133,11 @@ public Object run() {
       }
       DIRECT_BUFFER_CONSTRUCTOR = directBufferConstructor;
     } catch (Throwable e) {
-      throw new RuntimeException("Failed to initialize MemoryUtil.", e);
+      // This exception will get swallowed, but it's necessary for the static analysis that ensures
+      // the static fields above get initialized
+      final RuntimeException failure = new RuntimeException("Failed to initialize MemoryUtil", e);
+      failure.printStackTrace();
```

Comment:
Should we redirect the stacktrace to logger.error?

Reply:
I wanted this to fail as loudly as possible since Arrow just won't work without it

Reply:
Most cases this won't be an issue since console output is also redirected to logs. Only a very small percentage who miss redirecting console output to logs shall have to look further.

Reply:
Interesting, in my experience only explicit logs made it to the logfiles (and java.util.logging wasn't consistently redirected)…though maybe that's not a great example

Reply:
In my experience, we always had System error/out redirected to a log file and the logger log file is also there. Eventually all the logs make it into Splunk.
Anyways nothing major, we can always refine later. Let's plan to merge the PR once committers approve